chrome/content/zotero/zoteroPane.js

```diff
@@ -175,14 +167,60 @@ var ZoteroPane = new function()
 		const buttons = toolbar.getElementsByTagName("toolbarbutton");
 		const focusMap = new Map();
 		const zones = [
-			new TbFocusZone(
-				"zotero-tb-collection-add", 
-				null, 
-				"zotero-tb-search-menu-button"),
-			new TbFocusZone(
-				"zotero-tb-locate", 
-				"zotero-tb-search",
-				null)
+			{
+				get start() { return document.getElementById("zotero-tb-collection-add"); },
+				focusBefore() {
+					// need to check if the itembox is open
+					const pane = document.getElementById("zotero-item-pane-content");
+					if (pane.selectedIndex === "0") {
+						document.getElementById("item-tree-main-default").focus();
+					}
+					else {
+						const tabBox = document.getElementById("zotero-view-tabbox");
+						if (tabBox.selectedIndex === 0) {
+							const itembox = document.getElementById("zotero-editpane-item-box");
+							itembox.focusLastField();
+						}
+						else if (tabBox.selectedIndex === 1) {
+							const notes = document.getElementById("zotero-editpane-notes");
+							const nodes = notes.querySelectorAll("button");
+							const node = nodes[nodes.length - 1];
+							node.focus();
+							// TODO: the notes are currently inaccessible to the keyboard
+						}
+						else if (tabBox.selectedIndex === 2) {
+							const tagContainer = document.getElementById("tags-box-container");
+							const tags = tagContainer.querySelectorAll("#tags-box-add-button,.zotero-clicky");
+							const next = tags[tags.length - 1];
+							if (next.id === "tags-box-add-button") {
+								next.focus();
+							}
+							else {
+								next.click();
+							}
+						}
+						else if (tabBox.selectedIndex === 3) {
+							const related = tabBox.querySelector("relatedbox");
+							related.receiveKeyboardFocus();
```

Comment:
If we're going to bother with a function instead of reaching in directly, we should pass in a direction, even if it's not used now.